### PR TITLE
[RLlib][RLModule] Fix minibatch cyclic iterator bug

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Why are these changes needed?
This PR fixes some bug in minibatch cyclic iterator that produced infinit loops. It also adds a unittest to hedge against future regressions. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/minibatch_utils.py [hunk 1]
```diff
@@ -59,17 +59,20 @@ def __iter__(self):
...
                 s = self._start[module_id]  # start
-                e = s + self._minibatch_size  # end
+                n_steps = self._minibatch_size
 
                 samples_to_concat = []
                 # cycle through the batch until we have enough samples
-                while e >= len(module_batch):
-                    samples_to_concat.append(module_batch[s:])
-                    e = self._minibatch_size - len(module_batch[s:])
+                while n_steps >= len(module_batch[s:]):
```
Comment (avnishn) [inline rllib/utils/minibatch_utils.py:66]: how does this handle the base case of module_batch having 0 timesteps

**Review by avnishn — COMMENTED** [review]
one small comment. I think that the code may error / hang in the case that a module_batch is of size 0

File: rllib/utils/minibatch_utils.py (hunk 1 above)
Comment (gjoliver) [reply rllib/utils/minibatch_utils.py:66]: module_batch[s:] will create a new list for each iteration.
small performance penalty that may be unnecessary.
also, we can stay with the s and e model I feel like, just need to flip the >= to < right? :)

Comment (avnishn) [reply rllib/utils/minibatch_utils.py:66]: if module_batch is empty, then we will never exit this while loop, unless n_steps is 0, or negative, AFAICT

Comment (kouroshHakha) [reply rllib/utils/minibatch_utils.py:66]: > how does this handle the base case of module_batch having 0 timesteps
module_batch should never be empty. Logically that should actually create an infinite loop. I will add a check for this.

Comment (kouroshHakha) [reply rllib/utils/minibatch_utils.py:66]: > module_batch[s:] will create a new list for each iteration.

Fixed it via `while n_steps >= len(module_batch) - s`

Comment (kouroshHakha) [reply rllib/utils/minibatch_utils.py:66]: > also, we can stay with the s and e model I feel like, just need to flip the >= to < right? :)

This way of writing it was more intuitive to me at least. After I figured out the cause for infinit loop.

Comment (kouroshHakha) [reply rllib/utils/minibatch_utils.py]: it's done. We do now.

Comment (avnishn) [reply rllib/utils/minibatch_utils.py]: yay

**Review by gjoliver — APPROVED** [review]

**Pull request merged**